src/pages/home/report/ReportActionContextMenu.js

```diff
@@ -57,14 +22,57 @@ const propTypes = {
 
     // Controls the visibility of this component.
     isVisible: PropTypes.bool,
+
+    // Function to trigger when we try to mark a message as unread
+    onMarkAsUnread: PropTypes.func,
 };
 
 const defaultProps = {
     isMini: false,
     isVisible: false,
+    onMarkAsUnread: () => {},
 };
 
 const ReportActionContextMenu = (props) => {
+    /**
+     * A list of all the context actions in this menu.
+     */
+    const CONTEXT_ACTIONS = [
```

Comment:
NAB, one small side effect of this change is that we are now recreating this object with each render. I don't think it will cause problems tho.

Reply:
Rather than recreating this object with every render, I think we should convert this to a class component, make `contextActions` a property of the class, then set it once in `componentDidMount`

Reply:
Or the `constructor()`